packages/components/pages/f-loyalty/src/components/_tests/Slots.test.js

```diff
@@ -0,0 +1,210 @@
+import { createLocalVue, mount } from '@vue/test-utils';
+import VueI18n from 'vue-i18n';
+import {
+    StampCard1 as StampCard,
+    StampCardPromotionCard
+} from '@justeat/f-content-cards';
+import Slots from '../Slots.vue';
+import i18nMocker from './helper';
+import tenantConfigs from '../../tenants';
+import NoCardsErrorState from '../NoCardsErrorState.vue';
+
+// MOCKS
+const MOCK_LOCALE = 'en-GB';
+const MOCK_CARDS = [
+    { id: 1, type: 'Stamp_Card_1', isReadyToClaim: true },
+    { id: 2, type: 'Stamp_Card_1', isReadyToClaim: true },
+    { id: 3, type: 'Stamp_Card_1', isReadyToClaim: true }
+];
+const MOCK_CARDS_IN_PROGRESS = [
+    { id: 1, type: 'Stamp_Card_1', isReadyToClaim: false },
+    { id: 2, type: 'Stamp_Card_1', isReadyToClaim: false },
+    { id: 3, type: 'Stamp_Card_1', isReadyToClaim: false }
+];
+const MOCK_CARDS_PARTICIPATING_RESTAURANTS = [
+    { id: 1, type: 'StampCard_Promotion_Card_1' },
+    { id: 2, type: 'StampCard_Promotion_Card_1' },
+    { id: 3, type: 'StampCard_Promotion_Card_1' }
+];
+const MOCK_PROMOTION_CARD_ID = '__TEST_PROMOTION_CARD_ID__';
+const MOCK_STAMP_CARD_ID = '__TEST_STAMP_CARD_ID__';
+
+const localVue = createLocalVue();
+localVue.use(VueI18n);
+
+const wrapperOptions = {
+    stubs: {
+        StampCard: { template: `<div data-test-id="${MOCK_STAMP_CARD_ID}">TEST_STAMP_CARD</div>` },
+        StampCardPromotionCard: { template: `<div data-test-id="${MOCK_PROMOTION_CARD_ID}">TEST_PROMOTION_CARD</div>` }
+    },
+    mocks: {
+        $t: key => i18nMocker(key, MOCK_LOCALE)
+    }
+};
+
+let wrapper;
+let zeroCardsWrapper;
+
+describe('Slots.vue', () => {
+    beforeEach(() => {
+        // Arrange
+        zeroCardsWrapper = mount(Slots, {
+            localVue,
+            propsData: {
+                cards: []
+            },
+            ...wrapperOptions
+        });
+    });
+
+    describe('Redeemable Stamp Cards', () => {
+        beforeEach(() => {
+            // Arrange
+            wrapper = mount(Slots, {
+                localVue,
+                propsData: {
+                    cards: MOCK_CARDS
+                },
+                ...wrapperOptions
+            });
+        });
+
+        it('should display the readyToClaimTitle when hasRedeemableStampcards is true', () => {
+            // Act
+            const title = wrapper.find('[data-test-id="section-readyToClaim-title"]');
```

Comment:
Gap for readability:
```suggestion
            const title = wrapper.find('[data-test-id="section-readyToClaim-title"]');

```